roachtest: get dmesg on failed runs

tbg opened the pull request:
This helps verify oom killer activity, such as presumably seen in
\#34113.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/34217)
<!-- Reviewable:end -->

petermattis reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> modulo question about extraneous parameter.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/34217#-:-LX-GoH96vtukLG9VezH:bgp9m25)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @tbg)

___
*[pkg/cmd/roachtest/cluster.go, line 946 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/34217#-LX-Gg2GAFWdOoVkOSVG:-LX-Gg2GAFWdOoVkOSVH:bm33a97) ([raw file](https://github.com/cockroachdb/cockroach/blob/66536f92480c0751644aad777f97976e7340bc83/pkg/cmd/roachtest/cluster.go#L946)):*
> ```Go
> 		return err
> 	}
> 	if err := execCmd(execCtx, c.l, roachprod, "ssh", c.name, "--", "/bin/bash", "-c", "'sudo dmesg > "+name+"'" /* src */, path /* dest */); err != nil {
> ```

What is `path` doing in this command? Seems extraneous.


<!-- Sent from Reviewable.io -->

petermattis reviewed (commented):
We should probably condition this `dmesg` fetch to only occur for non-local runs. `sudo dmesg` on my laptop will ask for a password. Maybe it just fails due to not having standard in. 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/34217#-:-LX-GyhNDJepIX0JMWMF:buv6pb4)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @tbg)



<!-- Sent from Reviewable.io -->

tbg committed:
roachtest: get dmesg on failed runs

This helps verify oom killer activity, such as presumably seen in
\#34113.

Release note: None

tbg commented:
Done and done. TRTR!

bors r=petermattis
